File: Java/PayrollSystem/src/main/java/SqlRepository.java
import java.io.Closeable;
import java.io.IOException;
import java.util.Arrays;

/**
 * Created by astral on 24.03.15.
 */
public class SqlRepository implements Closeable {
  public void Dispose() {

  }

  public Iterable<Employee> CurrentEmployees() {
    return Arrays.asList(new Employee(), new Employee(), new Employee());
  }

  @Override
  public void close() throws IOException {
    System.out.println("Disposing");
  }
}

File: CSharp/TellDontAsk/PayrollSystem/SqlRepository.cs
using System;
using System.Collections.Generic;

namespace PayrollSystem
{
  public class SqlRepository : IDisposable
  {
    public void Dispose()
    {
      Console.WriteLine("Disposing");
    }

    public IEnumerable<Employee> CurrentEmployees()
    {
      return new[] { new Employee(), new Employee(), new Employee() };
    }
  }

}

